blocks/filesystem/base.py

```diff
@@ -1,18 +1,45 @@
-from abc import ABC, abstractmethod
+from typing import Union, Sequence, Tuple, List
+from collections import defaultdict
+from fsspec.core import split_protocol, get_filesystem_class, has_magic
 
-from typing import Union, IO, Sequence, Iterator, List, Any, Generator, ContextManager
-from blocks.datafile import DataFile
-from contextlib import contextmanager
 
+class FileSystem:
```

Comment:
super nit: this class name is now a bit of a misnomer since it can contain multiple fsspec filesystems